File: app/src/main/java/com/dynamo/spacex/SpaceXListAdapter.java
package com.dynamo.spacex;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.dynamo.spacex.databinding.ItemSpacexViewHolderBinding;
import com.dynamo.spacex.viewmodels.SpaceXListViewModel;
import com.dynamo.spacex.viewmodels.SpaceXViewModel;

import java.util.List;

public class SpaceXListAdapter extends RecyclerView.Adapter<SpaceXItemViewHolder> {

    private List<SpaceXViewModel> spaceXViewModels;
    private SpaceXListViewModel spaceXListViewModel;

    public SpaceXListAdapter(List<SpaceXViewModel> spaceXViewModels, SpaceXListViewModel spaceXListViewModel) {
        this.spaceXViewModels = spaceXViewModels;
        this.spaceXListViewModel = spaceXListViewModel;
    }

    @NonNull
    @Override
    public SpaceXItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        ItemSpacexViewHolderBinding spacexViewHolderBinding = ItemSpacexViewHolderBinding.inflate(layoutInflater, parent, false);
        return new SpaceXItemViewHolder(spacexViewHolderBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull SpaceXItemViewHolder holder, int position) {
        SpaceXViewModel viewModel = spaceXViewModels.get(position);
        holder.bind(viewModel, spaceXListViewModel);
    }

    @Override
    public int getItemCount() {
        return (spaceXViewModels != null ? spaceXViewModels.size() : 0);
    }
}

File: app/src/main/java/com/dynamo/spacex/SpaceXListAdapter.kt
package com.dynamo.spacex

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.dynamo.spacex.databinding.ItemSpacexViewHolderBinding
import com.dynamo.spacex.viewmodels.SpaceXListViewModel
import com.dynamo.spacex.viewmodels.SpaceXViewModel

class SpaceXListAdapter (private val spaceXListViewModel: SpaceXListViewModel): RecyclerView.Adapter<SpaceXItemViewHolder>() {

    private val spaceXViewModels = ArrayList<SpaceXViewModel>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SpaceXItemViewHolder {
        val layoutInflater: LayoutInflater = LayoutInflater.from(parent.context)
        val spacexViewHolderBinding: ItemSpacexViewHolderBinding = ItemSpacexViewHolderBinding.inflate(layoutInflater, parent, false)
        return SpaceXItemViewHolder(spacexViewHolderBinding)
    }

    override fun getItemCount(): Int {
        return spaceXViewModels.size
    }

    override fun onBindViewHolder(holder: SpaceXItemViewHolder, position: Int) {
        val viewModel: SpaceXViewModel = spaceXViewModels[position]
        holder.bind(viewModel, spaceXListViewModel)
    }

    fun setSpaceXViewModels(spaceXViewModels: List<SpaceXViewModel>) {
        this.spaceXViewModels.addAll(spaceXViewModels)
        notifyDataSetChanged()
    }
}